libraries/package-deps-hash/src/getPackageDeps.ts

```diff
@@ -2,10 +2,41 @@
 // See LICENSE in the project root for license information.
 
 import * as child_process from 'child_process';
+import * as path from 'path';
 import { Executable } from '@rushstack/node-core-library';
 
 import { IPackageDeps } from './IPackageDeps';
 
+/**
+ * Parses a quoted filename sourced from the output of the "git status" command.
+ *
+ * Paths with non-standard characters will be enclosed with double-quotes, and non-standard
+ * characters will be backslash escaped (ex. double-quotes, non-ASCII characters). The
+ * escaped chars can be included in one of two ways:
+ * - backslash-escaped chars (ex. \")
+ * - octal encoded chars (ex. \347)
+ *
+ * See documentation: https://git-scm.com/docs/git-status
+ */
+export function parseGitFilename(filename: string): string {
+  // If there are no double-quotes around the string, then there are no escaped characters
+  // to decode, so just return
+  if (!filename.match(/^".+"$/)) {
+    return filename;
+  }
+
+  // Need to hex encode '%' since we will be decoding from hex
+  filename = filename.replace(/%/g, '%25');
+  // Replace all instances of octal encoding with hex (ex. '\347\275\221' -> '%E7%BD%91')
```

Comment:
This code comment should clarify that Git's "octal" literals represent UTF8 bytes, so we convert them to percent-encoded bytes, because `decodeURIComponent()` can decode that into Unicode characters.

(From reading this code, I had mistakenly inferred that Git's output could include percent-encoded characters.)